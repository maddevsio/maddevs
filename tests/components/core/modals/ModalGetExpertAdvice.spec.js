import { render, fireEvent, screen } from '@testing-library/vue'
import ModalGetExpertAdvice from '@/components/core/modals/ModalGetExpertAdvice'
import formBaseProps from '../../../__mocks__/formBaseProps'
import delay from '../../../../client/helpers/delay'

const mocks = formBaseProps
mocks.$v.questionsOnItConsulting = {
  $touch: jest.fn(),
}

describe('ModalGetExpertAdvice component', () => {
  it('should render correctly', () => {
    const { container } = render(ModalGetExpertAdvice, {
      mocks,
    })

    expect(container).toMatchSnapshot()
  })

  it('should correct call handler', async () => {
    render(ModalGetExpertAdvice, {
      mocks,
    })

    await fireEvent.update(screen.getByTestId('test-base-form-phone'), '+3 4546-657-9098-7875645342245')
    await fireEvent.update(screen.getByTestId('test-base-form-email'), 'someemail@maddevs.io')
    await fireEvent.update(screen.getByTestId('test-base-form-full-name'), 'John Johnson')

    await fireEvent.click(screen.queryByTestId('test-privacy-policy-checkbox-input'))
    await fireEvent.click(screen.getByText('Get expert advice'))

    await delay(1000)
    // Called in BaseForm and GetExpertAdviceForm components
    expect(mocks.$v.$reset).toHaveBeenCalledTimes(2)
  })
})
