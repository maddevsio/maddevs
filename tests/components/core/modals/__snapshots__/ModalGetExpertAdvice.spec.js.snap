// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalGetExpertAdvice component should render correctly 1`] = `
<div>
  <div
    class="form"
  >
    <div
      class="fields-list"
    >
      <div
        class="modal-field-item field-item"
      >
        <p
          class="modal-field-name field-name required"
        >
          
    Full Name
  
        </p>
         
        <div>
          <input
            class="modal-entry-field entry-field"
            data-testid="test-base-form-fullName"
            placeholder="John Smith"
            type="text"
          />
           
          <!---->
        </div>
         
        <!---->
      </div>
       
      <!---->
       
      <div
        class="modal-field-item field-item"
      >
        <p
          class="modal-field-name field-name required"
        >
          
    Work email
  
        </p>
         
        <div>
          <input
            class="modal-entry-field entry-field"
            data-testid="test-base-form-email"
            placeholder="your@mail.com"
            type="text"
          />
           
          <!---->
        </div>
         
        <!---->
      </div>
       
      <div
        class="modal-field-item field-item"
      >
        <p
          class="modal-field-name field-name"
        >
          
    Phone number
  
        </p>
         
        <div>
          <input
            class="modal-entry-field entry-field"
            data-testid="test-base-form-phoneNumber"
            placeholder="+X XXX XXX-XXXX"
            type="text"
          />
           
          <!---->
        </div>
         
        <!---->
      </div>
       
      <!---->
       
      <div
        class="modal-field-item field-item"
      >
        <p
          class="modal-field-name field-name"
        >
          
    Your questions on IT consulting
  
        </p>
         
        <div>
          <!---->
           
          <textarea
            class="modal-entry-field entry-field textarea"
            data-testid="test-base-form-questionsOnItConsulting"
            placeholder="Tell us how we can help you…"
          />
        </div>
         
        <!---->
      </div>
    </div>
     
    <div
      class="form-checkboxes"
    >
      <label
        class="form-checkbox-label required"
        for="privacy-policy-get-expert-advice-form"
      >
        I confirm that I have read and accepted
    
        <div>
          Mad Devs’
        </div>
         
        <a
          href="/privacy/"
          target="_blank"
        >
          Privacy Policy
        </a>
         
        <input
          class="form-checkbox-input"
          data-testid="test-privacy-policy-checkbox-input"
          id="privacy-policy-get-expert-advice-form"
          name="Privacy policy"
          type="checkbox"
        />
         
        <div
          class="checkmark"
        />
      </label>
       
      <label
        class="form-checkbox-label"
        for="marketing-communications-get-expert-advice-form"
      >
        I agree to get
    
        <div>
          Mad Devs’
        </div>
        
    discount offers and other marketing communications.
    
        <input
          class="form-checkbox-input"
          data-testid="test-form-checkbox-input"
          id="marketing-communications-get-expert-advice-form"
          name="Marketing communications"
          type="checkbox"
        />
         
        <div
          class="checkmark"
        />
      </label>
    </div>
     
    <button
      class="ui-button ui-button--disabled modal-button"
    >
      
    Get expert advice
  
    </button>
  </div>
</div>
`;
